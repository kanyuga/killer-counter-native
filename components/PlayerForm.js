import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, TextInput, Button } from 'react-native';


export default class PlayerForm extends Component {
    constructor(props) {
        super(props);
        this.state = {name: ''};
    }

    handlePress = () => {
        this.props.onSubmit(this.state.name);
        this.setState({ name: '' });
    };

    render () {
        return (
            <View style={{flexDirection: 'row', width : '100%'}}>
                <TextInput
                    style={{flex: 3, paddingVertical: 0}}
                    placeholder="Enter Player Name"
                    value={this.state.name}
                    onChangeText={(name) => this.setState({name})}
                    onSubmitEditing={this.handlePress}
                    returnKeyType="done"
                    autoFocus
                />
                <Button style={{flex: 2}}
                        onPress={this.handlePress}
                        title="Add"
                />
            </View>
        )
    }
}

PlayerForm.propTypes = {
    onSubmit: PropTypes.func.isRequired
};