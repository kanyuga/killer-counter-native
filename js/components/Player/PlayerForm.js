import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View, TextInput} from 'react-native';
import AppButton from "../common/AppButton";


const PlayerForm = (props) => (
  <View style={{display: 'flex', flexDirection: 'column', width: '100%'}}>
    <TextInput
      style={{paddingVertical: 10, color: '#ccc'}}
      placeholder="Enter Player Name"
      placeholderTextColor="#aaa"
      value={props.name}
      onChangeText={props.handleChange}
      onSubmitEditing={props.onSubmit}
      blurOnSubmit={false}
      returnKeyType="done"
      autoFocus
    />
    <View style={{display: 'flex', flexDirection: 'row', width: '100%', justifyContent: 'space-around'}}>
      <View style={{flex: 1, marginRight: 2}}>
        <AppButton
          onPress={props.onSubmit}
          title="Add Player"
          disabled={!props.name.trim()}
        />
      </View>
      {!props.started && props.startable &&
      <View style={{flex: 1, marginLeft: 2}}>
        <AppButton
          onPress={props.startGame}
          title="Start Game"
        />
      </View>
      }
    </View>
  </View>
);

PlayerForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  startGame: PropTypes.func.isRequired,
  started: PropTypes.bool.isRequired,
  startable: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired
};

export default PlayerForm;