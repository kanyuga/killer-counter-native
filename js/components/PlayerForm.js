import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, TextInput} from 'react-native';
import {connect} from "react-redux";
import {addPlayer, startGame} from "../actions";
import AppButton from "./AppButton";


class PlayerForm extends Component {
    constructor(props) {
        super(props);
        this.state = {name: ''};
    }

    handlePress = () => {
        this.props.onSubmit(this.state.name);
        this.setState({ name: '' });
    };

    render () {
        return (
            <View style={{display: 'flex', flexDirection: 'column', width : '100%'}}>

              <TextInput
                  style={{paddingVertical: 10, color: '#ccc'}}
                  placeholder="Enter Player Name"
                  placeholderTextColor="#aaa"
                  value={this.state.name}
                  onChangeText={(name) => this.setState({name})}
                  onSubmitEditing={this.handlePress}
                  blurOnSubmit={false}
                  returnKeyType="done"
                  autoFocus
                />
                <View style={{display: 'flex', flexDirection: 'row', width : '100%', justifyContent: 'space-around'}}>
                  <View style={{ flex: 1, marginRight: 2}}>
                    <AppButton
                      onPress={this.handlePress}
                      title="Add Player"
                      disabled={!this.state.name.trim()}
                    />
                  </View>
                  { !this.props.started && this.props.startable &&
                    <View style={{ flex: 1, marginLeft: 2}}>
                      <AppButton
                        onPress={this.startGame}
                        title="Start Game"
                      />
                    </View>
                  }
                </View>
            </View>
        )
    }
}

PlayerForm.propTypes = {
    onSubmit: PropTypes.func.isRequired
};

PlayerForm = connect(
  state => ({ started: state.started, startable: state.players.length > 1}),
  dispatch => ({
    onSubmit: (name) => dispatch(addPlayer(name)),
    startGame: () => dispatch(startGame())
  }))
(PlayerForm);

export default PlayerForm;