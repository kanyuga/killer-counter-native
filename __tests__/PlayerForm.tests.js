import React from 'react';
import PlayerForm from '../js/components/Player/PlayerForm';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

describe("snapshots", () => {
  it ("doesn't show the start button when not started or startable", () => {
    expect(renderer.create(
      <PlayerForm handleChange={() => {}} onSubmit={() => {}} startGame={() => {}} started={false} startable={false} name=""/>
    )).toMatchSnapshot();
  });
  it ("doesn't show the start button when started", () => {
    expect(renderer.create(
      <PlayerForm handleChange={() => {}} onSubmit={() => {}} startGame={() => {}} started={true} startable={false} name=""/>
    )).toMatchSnapshot();
  });
  it ("shows the start button when startable but not started", () => {
    expect(renderer.create(
      <PlayerForm handleChange={() => {}} onSubmit={() => {}} startGame={() => {}} started={false} startable={true} name=""/>
    )).toMatchSnapshot();
  });
});

describe("rendering", () => {
  let name = '';
  const onSubmit = (n) => name = n;
  it ("doesn't show the start button when not started or startable", () => {
    let playerForm = shallow(
      <PlayerForm handleChange={handleChange} onSubmit startGame={} started={true} startable={true} name="afafda"/>
    )
  });
});