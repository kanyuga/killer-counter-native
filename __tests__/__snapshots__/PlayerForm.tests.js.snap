// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering doesn't show the start button when not started or startable 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <TextInput
    autoFocus={true}
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Enter Player Name"
    placeholderTextColor="#aaa"
    returnKeyType="done"
    style={
      Object {
        "color": "#ccc",
        "paddingVertical": 10,
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginRight": 2,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#1B2836",
                "display": "flex",
                "justifyContent": "center",
                "padding": 10,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                },
                Object {
                  "color": "#ccc",
                },
              ]
            }
          >
             
            Add Player
             
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`rendering doesn't show the start button when started 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <TextInput
    autoFocus={true}
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Enter Player Name"
    placeholderTextColor="#aaa"
    returnKeyType="done"
    style={
      Object {
        "color": "#ccc",
        "paddingVertical": 10,
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginRight": 2,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#1B2836",
                "display": "flex",
                "justifyContent": "center",
                "padding": 10,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                },
                Object {
                  "color": "#ccc",
                },
              ]
            }
          >
             
            Add Player
             
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`rendering shows the start button when startable but not started 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <TextInput
    autoFocus={true}
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Enter Player Name"
    placeholderTextColor="#aaa"
    returnKeyType="done"
    style={
      Object {
        "color": "#ccc",
        "paddingVertical": 10,
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginRight": 2,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#1B2836",
                "display": "flex",
                "justifyContent": "center",
                "padding": 10,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                },
                Object {
                  "color": "#ccc",
                },
              ]
            }
          >
             
            Add Player
             
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 2,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#1B2836",
                "display": "flex",
                "justifyContent": "center",
                "padding": 10,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                },
                Object {
                  "color": "#ccc",
                },
              ]
            }
          >
             
            Start Game
             
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
